### PR TITLE
Share OpenSSL Crypto Implementation between simulator and tests

### PR DESCRIPTION
**Pull request by jhand2**
Make a crate with shared OpenSSL helpers to be used by both the DPE simulator and tests for the DPE crate.

Additionally, add support in the Crypto trait and implementations for hashing and ECDSA key generation.

### CODE REVIEW

File: simulator/src/main.rs [hunk 1]
```diff
@@ -133,20 +137,64 @@ fn main() -> std::io::Result<()> {
...
+
+    fn derive_cdi(
+        profile: DpeProfile,
+        measurement_digest: &[u8],
+        info: &[u8],
+    ) -> Result<Self::Cdi, DpeErrorCode> {
+        let md = match profile {
+            DpeProfile::P256Sha256 => MessageDigest::sha256(),
+            DpeProfile::P384Sha384 => MessageDigest::sha384(),
+        };
```
Comment (zhalvorsen) [inline simulator/src/main.rs:168]: It looks like we could do a helper function here. Do you think its worth it?

```Rust
impl SimCrypto {
    fn get_digest(profile: DpeProfile) -> MessageDigest {
        match profile {
            DpeProfile::P256Sha256 => MessageDigest::sha256(),
            DpeProfile::P384Sha384 => MessageDigest::sha384(),
        }
    }
}
```

File: ossl_crypto/Cargo.toml
[the remark this answers is not in the document]
Comment (jhand2) [reply ossl_crypto/Cargo.toml:1]: Ya, we should do them here too probably. caliptra-sw does.

File: simulator/src/main.rs (hunk 1 above)
Comment (jhand2) [reply simulator/src/main.rs:168]: Good idea. Done.

File: dpe/src/crypto.rs [hunk 2]
```diff
@@ -22,31 +62,112 @@ pub trait Crypto {
...
+    ///
+    /// # Arguments
+    ///
+    /// * `profile` - Which profile is being used. This will tell the platform
+    ///   which algorithm to use
+    /// * `measurement_digest` - A digest of the measurements which should be
+    ///   used for CDI derivation
+    /// * `info` - Caller-supplied info string to use in CDI derivation
+    fn derive_cdi(
+        profile: DpeProfile,
```
[the remark this answers is not in the document]
Comment (zhalvorsen) [reply dpe/src/crypto.rs:77]: Meh, I'm second guessing this comment while we think about how we want to handle profiles in the future. Marking as resolved.

**Review by zhalvorsen — APPROVED** [review]

**Pull request merged**